core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractValueFactory.java

```diff
@@ -0,0 +1,280 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ ******************************************************************************/
+
+package org.eclipse.rdf4j.model.base;
+
+import static org.eclipse.rdf4j.model.base.AbstractLiteral.reserved;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+
+import org.eclipse.rdf4j.model.BNode;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Triple;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.base.AbstractBNode.GenericBNode;
+import org.eclipse.rdf4j.model.base.AbstractIRI.GenericIRI;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.BooleanLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.CalendarLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.DecimalLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.IntegerLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.NumberLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.TaggedLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.TypedLiteral;
+import org.eclipse.rdf4j.model.base.AbstractStatement.GenericStatement;
+import org.eclipse.rdf4j.model.base.AbstractTriple.GenericTriple;
+
+/**
+ * Base class for {@link ValueFactory}, offering common functionality.
+ *
+ * @author Alessandro Bollini
+ * @since 3.5.0
+ */
+@SuppressWarnings("UseOfObsoleteDateTimeApi")
+public abstract class AbstractValueFactory implements ValueFactory {
```

Comment:
I've just realized that this new class was introduced, but there's no relation to the existing AbstractValueFactory. I'll take a look at that, we may want to have the old class inherit from the new and/or deprecate it.

Reply:
See #???